### ISSUE TITLE
MD024 incorrectly reports different headings as duplicates

### ISSUE THREAD

**Issue by 3C0D** [DavidAnson/vscode-markdownlint#404]
**What did you do?**
I created two level 4 headings in my Markdown file that appear to have different content.

**What did you expect to happen?**
No MD024 violation should be reported since the headings have different text content.

**What actually happened?**
MD024/no-duplicate-heading error was triggered, claiming both headings have the same content even though they visibly differ:
- Line 79: `#### Sans source maps`
- Line 88: `#### Avec source maps`

**Error message:**
```
MD024/no-duplicate-heading: Multiple headings with the same content
```

**How can the issue be reproduced?**
Create a Markdown file with these two headings at different locations:
```markdown
#### Sans source maps

#### Avec source maps
```

**Version information:**
- markdownlint version: [0.60.0]
- Operating system: [windows 11]

**Additional context:**
The headings contain different text but MD024 incorrectly identifies them as duplicates. They both contain emoji/special characters (🎨 icon visible in the screenshot).

<img width="651" height="480" alt="Image" src="https://github.com/user-attachments/assets/e185e305-113e-463b-867d-d1a04d9ef4f0" />

**Comment by 3C0D** [comment]
This change could be quite good to have as the default parameter. And really, if we want to remove doing:
    "MD024": {
      "siblings_only": false
    }


But OK, maybe that would break the logic of the other commands. Maybe it would be good to have a default configuration that limits the number of false positives. Because that's not the kind of error we want to correct. So having a pre-prepared configuration, maybe as an example or I don't know.

**Comment by DavidAnson** [comment]
The rule is called "no-duplicate-heading", so that's what it enforces by default. If you want some duplicates, you can opt in.
